In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# raw_data = pd.read_csv(r'E:\Nextstep\Scaler\Business-Case-Study\Aerofit\Dataset\aerofit_treadmill.csv')
raw_data = pd.read_csv(r'F:\Muthu_2023\Personal\NextStep\DSCourse\Scaler\Business-Case-Study\Aerofit\Dataset\aerofit_treadmill.csv')

In [ ]:
raw_data.head()

In [ ]:
raw_data.info()

**Inference:**
- `No null values in all the columns`
- `3 categorical and 5 numerical columns`

In [ ]:
raw_data.describe()

**Inference:**
- `25% of users in the age range of 24 to 26`
- `Education, Usage and fitness seems to be integer and it requires detailed analysis`
- ``

In [ ]:
raw_data.describe(include='object')

# Exploratory Data Analysis

In [ ]:
plt.rcParams['figure.figsize'] = (12,6)

## UVA on Categorical column

In [ ]:
def plot_count(df, col):
    plt.suptitle('Sales Quantity by ' + col)
    plt.subplot(1,2,1)
    sns.countplot(data = df, x = col)
    plt.subplot(1,2,2)
    plt.pie(df[col].value_counts(), labels = list(df[col].value_counts().index), autopct = '%0.1f%%')

In [ ]:
plot_count(raw_data, 'Product')

**Inference:**
- `KP281 has highest market share among other products`

In [ ]:
plot_count(raw_data, 'Gender')

**Inference:**
- `Purchase of male customer is the most`

In [ ]:
plot_count(raw_data, 'MaritalStatus')

**Inference:**
- `Partnered makes the most purchase than single`

In [ ]:
plot_count(raw_data, 'Usage')

**Inferece:**
- `Users purchased the most who uses the treadmill 3-4 days a week`
- `80% of users uses the treadmill 2-4 days a week`

In [ ]:
plot_count(raw_data, 'Fitness')

**Inference:**
- `Average rated users are the most in purchasing the treadmill`

In [ ]:
plot_count(raw_data, 'Education')

**Inference:**
- `Users with 14 and 16 years of education purchases the most`
- `Need further investigation on this part how it relates to the purchase of treadmill`

## UVA on Numerical columns

In [ ]:
def plot_hist(df, col):
    plt.suptitle('Sales Quantity by ' + col)
    plt.subplot(1,2,1)
    sns.histplot(data = df, x = col)
    plt.subplot(1,2,2)
    sns.boxplot(data = df, y = col)
    plt.show()

In [ ]:
plot_hist(raw_data, 'Age')

**Inference:**
- `Predominant of the purchase is peformed by 20 to 35 age groups`

In [ ]:
plot_hist(raw_data, 'Income')

**Inference:**
- `75% of the purchases are performed by users with income lesser than 60`

In [ ]:
plot_hist(raw_data, 'Miles')

**Inference:**
- `Users who runs 50-100 miles are the most who shown the interest in purchasing the treadmill`

# Bivariate Analysis

In [ ]:
plt.figure(figsize=(12, 16))
plt.subplot(3,2,1)
sns.countplot(data = raw_data, hue = 'Product', x='Gender')
plt.subplot(3,2,2)
sns.countplot(data = raw_data, hue = 'Product', x='MaritalStatus')
plt.subplot(3,2,3)
sns.countplot(data = raw_data, hue = 'Product', x='Usage')
plt.subplot(3,2,4)
sns.countplot(data = raw_data, hue = 'Product', x='Fitness')
plt.subplot(3,1,3)
sns.countplot(data = raw_data, hue = 'Product', x='Education')

**Inference:**
- ***Product vs Gender:*** `Female show very less interest in buying the product KP781`
- ***Product vs Martial Status:*** `Partnered users are highly interested in buying treadmill`
- ***Product vs Usage:*** `Users with high usage are more prone to buy KP781 and less usage prefers other 2 products`
- ***Product vs Fitness:*** `Medium fitness users (2-4) are much interested in buying treadmill predominantly and only High fitness users prefer KP781`
- ***Product vs Education:*** `Higher education group prefers product KP781`

In [ ]:
plt.figure(figsize=(12, 6))
plt.subplot(1,3,1)
sns.boxplot(data = raw_data, x = 'Product', y='Age')
plt.subplot(1,3,2)
sns.boxplot(data = raw_data, x = 'Product', y='Income')
plt.subplot(1,3,3)
sns.boxplot(data = raw_data, x = 'Product', y='Miles')
# plt.subplot(3,2,4)
# sns.countplot(data = raw_data, hue = 'Product', x='Fitness')
# plt.subplot(3,1,3)
# sns.countplot(data = raw_data, hue = 'Product', x='Education')
plt.tight_layout()

**Inference:**
- ***Age vs Product:*** `Predominant of the users of KP781 are in 20s age group`
- ***Income vs Product:*** `Purchase of KP781 is dominated by higher income users. Hence it seems that KP781 is on costlier side.
KP481 is in the medium range and KP 281 is on the lower side`
- ***Miles vs Product:*** `Purchase of KP781 is dominated by higher miles users and it seems that it is preferred by heavy workout users`

# Multivariate Analysis

In [ ]:
sns.boxplot(data=raw_data, y = 'Age', x = 'Gender', hue='Product')

**Inference:**
- `Median age of male interested in KP481 is lesser than other 2 products whereas it is vice versa for female users`


In [ ]:
sns.boxplot(data=raw_data, y = 'Income', x = 'Gender', hue='Product')

In [ ]:
sns.boxplot(data=raw_data, y = 'Miles', x = 'Gender', hue='Product')

In [ ]:
sns.boxplot(data=raw_data, y = 'Age', x = 'MaritalStatus', hue='Product')

In [ ]:
sns.boxplot(data=raw_data, y = 'Age', x = 'Usage', hue='Product')

**Inference:**
- `High usage `

In [ ]:
sns.boxplot(data=raw_data, y = 'Income', x = 'Product', hue='Fitness')

In [ ]:
sns.jointplot(data=raw_data, x = 'Income',  y='Age', hue = 'Product')

# Conditional Probability

## Gender and Product

In [38]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins = True, normalize='columns')
print("\033[92m", 'Conditional Probability given Gender')
display(cp_df)

 Conditional Probability given Gender


Gender     Female      Male       All
Product                              
KP281    0.526316  0.384615  0.444444
KP481    0.381579  0.298077  0.333333
KP781    0.092105  0.317308  0.222222

**Inference:**
- `Above table represents the conditional probability of buying an item (row) given gender (column)`
- `For Ex: Probability of buying KP281 given female is 0.526316`
- `Table infers that Female are more prone to buy products KP281 and KP481 than KP781 compared to male`

In [37]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins = True, normalize='index')
print("\033[92m", 'Conditional Probability given Product')
display(cp_df)

 Conditional Probability given Product


Gender     Female      Male
Product                    
KP281    0.500000  0.500000
KP481    0.483333  0.516667
KP781    0.175000  0.825000
All      0.422222  0.577778

**Inference:**
- `Above table represents the conditional probability of being a gender for the given item`
- `For Ex: Probability of being Male given KP481 is bought, is 0.51667`
- `It indicates, both gender are almost equally likely to buy KP281 and KP481 products`

## Education and Product

In [41]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Education'], margins = True, normalize='columns')
print("\033[92m", 'Conditional Probability given Education')
display(cp_df)

 Conditional Probability given Education


Education        12   13        14   15        16        18   20   21  \
Product                                                                 
KP281      0.666667  0.6  0.545455  0.8  0.458824  0.086957  0.0  0.0   
KP481      0.333333  0.4  0.418182  0.2  0.364706  0.086957  0.0  0.0   
KP781      0.000000  0.0  0.036364  0.0  0.176471  0.826087  1.0  1.0   

Education       All  
Product              
KP281      0.444444  
KP481      0.333333  
KP781      0.222222

**Inference:**
- `Above table represents the conditional probability of buying an item (row) given educational years (column)`
- `For Ex: Probability of buying KP281 given educational years as 12, is 0.66667`
- `Table infers that users with 18 and more years of education tend buy product KP781 than KP281 and KP481`

In [42]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Education'], margins = True, normalize='index')
print("\033[92m", 'Conditional Probability given Product')
display(cp_df)

 Conditional Probability given Product


Education        12        13        14        15        16        18  \
Product                                                                 
KP281      0.025000  0.037500  0.375000  0.050000  0.487500  0.025000   
KP481      0.016667  0.033333  0.383333  0.016667  0.516667  0.033333   
KP781      0.000000  0.000000  0.050000  0.000000  0.375000  0.475000   
All        0.016667  0.027778  0.305556  0.027778  0.472222  0.127778   

Education        20        21  
Product                        
KP281      0.000000  0.000000  
KP481      0.000000  0.000000  
KP781      0.025000  0.075000  
All        0.005556  0.016667

**Inference:**
- `Above table represents the conditional probability of having respective educational year for the purchase of given item`
- `For Ex: Probability of having educational years of 16, given KP481 is bought, is 0.51667`
- `It indicates, if an item is bought user is most likely to be having the educational year of 16`

## Marital Status and Product

In [44]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['MaritalStatus'], margins = True, normalize='columns')
print("\033[92m", 'Conditional Probability given Marital Status')
display(cp_df)

 Conditional Probability given Marital Status


MaritalStatus  Partnered    Single       All
Product                                     
KP281           0.448598  0.438356  0.444444
KP481           0.336449  0.328767  0.333333
KP781           0.214953  0.232877  0.222222

**Inference:**
- `Above table represents the conditional probability of buying an item (row) given Marital Status (column)`
- `For Ex: Probability of buying KP281 given Partnered is 0.448598`
- `Table infers that both single and partnered are equally likely to buy any product`

In [45]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['MaritalStatus'], margins = True, normalize='index')
print("\033[92m", 'Conditional Probability given Product')
display(cp_df)

 Conditional Probability given Product


MaritalStatus  Partnered    Single
Product                           
KP281           0.600000  0.400000
KP481           0.600000  0.400000
KP781           0.575000  0.425000
All             0.594444  0.405556

**Inference:**
- `Above table represents the conditional probability of being in Marital status for the purchase of given item`
- `For Ex: Probability of being Partnered given KP481 is bought, is 0.6`
- `It indicates, if an item is bought the user is slightly more likely to be partnered`

## Usage and Product

In [47]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Usage'], margins = True, normalize='columns')
print("\033[92m", 'Conditional Probability given Usage')
display(cp_df)

 Conditional Probability given Usage


Usage           2         3         4         5    6    7       All
Product                                                            
KP281    0.575758  0.536232  0.423077  0.117647  0.0  0.0  0.444444
KP481    0.424242  0.449275  0.230769  0.176471  0.0  0.0  0.333333
KP781    0.000000  0.014493  0.346154  0.705882  1.0  1.0  0.222222

**Inference:**
- `Above table represents the conditional probability of buying an item (row) given weekly usage (column)`
- `For Ex: Probability of buying KP281 given weekly usage of 3 is 0.536232`
- `Table infers that if the weekly usage is less than or 4, the user is more likely to buy KP281 and KP481`
- `Especially if the usage is 6 or 7 days a week, the user will definitely go with KP781 (100% probability)`

In [48]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Usage'], margins = True, normalize='index')
print("\033[92m", 'Conditional Probability given Product')
display(cp_df)

 Conditional Probability given Product


Usage           2         3         4         5         6         7
Product                                                            
KP281    0.237500  0.462500  0.275000  0.025000  0.000000  0.000000
KP481    0.233333  0.516667  0.200000  0.050000  0.000000  0.000000
KP781    0.000000  0.025000  0.450000  0.300000  0.175000  0.050000
All      0.183333  0.383333  0.288889  0.094444  0.038889  0.011111

**Inference:**
- `Above table represents the conditional probability of having weekly usage given an item is purchased`
- `For Ex: Probability of having weekly usage of 3 given KP481 is bought, is 0.51667`
- `It indicates, if a user purchased KP281 and KP481 products then the user likely to be using it for 3 - 4 days a week`

## Fitness and Product

In [49]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Fitness'], margins = True, normalize='columns')
print("\033[92m", 'Conditional Probability given Fitness')
display(cp_df)

 Conditional Probability given Fitness


Fitness    1         2         3         4         5       All
Product                                                       
KP281    0.5  0.538462  0.556701  0.375000  0.064516  0.444444
KP481    0.5  0.461538  0.402062  0.333333  0.000000  0.333333
KP781    0.0  0.000000  0.041237  0.291667  0.935484  0.222222

**Inference:**
- `Above table represents the conditional probability of buying an item (row) given fitness (column)`
- `For Ex: Probability of buying KP281 given medium fitness (3) is 0.556701`
- `Table infers that if the rated fitness of user is less than 4, the user is highly likely to purchase products KP281 and KP481
- `Also if the user is rated a high fitness level (5), the user most likely to purchase KP781`

In [50]:
cp_df = pd.crosstab(raw_data['Product'], columns = raw_data['Fitness'], margins = True, normalize='index')
print("\033[92m", 'Conditional Probability given Product')
display(cp_df)

 Conditional Probability given Product


Fitness         1         2         3         4         5
Product                                                  
KP281    0.012500  0.175000  0.675000  0.112500  0.025000
KP481    0.016667  0.200000  0.650000  0.133333  0.000000
KP781    0.000000  0.000000  0.100000  0.175000  0.725000
All      0.011111  0.144444  0.538889  0.133333  0.172222

**Inference:**
- `Above table represents the conditional probability of having a fitness level for the given item purchased`
- `For Ex: Probability of having fitness level 3 given KP481 is bought, is 0.65`
- `It indicates, if a user purchase products KP281 or KP481, the user is highly likely to be having a fitness of 3`
- `if a user purchase products KP781, the user is very highly likely to be having a fitness of 5`

In [40]:
raw_data.columns

Index(['Product', 'Age', 'Gender', 'Education', 'MaritalStatus', 'Usage',
       'Fitness', 'Income', 'Miles'],
      dtype='object')

In [25]:
pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins=True)#, normalize=True)

Gender   Female  Male  All
Product                   
KP281        40    40   80
KP481        29    31   60
KP781         7    33   40
All          76   104  180

In [31]:
test = pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins = True, normalize='columns')
test

Gender     Female      Male       All
Product                              
KP281    0.526316  0.384615  0.444444
KP481    0.381579  0.298077  0.333333
KP781    0.092105  0.317308  0.222222

In [33]:
test = pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins = True, normalize='index')
test

Gender     Female      Male
Product                    
KP281    0.500000  0.500000
KP481    0.483333  0.516667
KP781    0.175000  0.825000
All      0.422222  0.577778

**Inference:**
- `Probability of buying product KP281 given female is 0.22`
- `Probability of buying a treadmill given male is 0.577`

In [ ]:
pd.crosstab(raw_data['Product'], columns = raw_data['Usage'], margins=True, normalize=True)

In [6]:
test

Gender     Female      Male
Product                    
KP281    0.222222  0.222222
KP481    0.161111  0.172222
KP781    0.038889  0.183333

In [7]:
test.apply(lambda x: x/x.sum())

Gender     Female      Male
Product                    
KP281    0.526316  0.384615
KP481    0.381579  0.298077
KP781    0.092105  0.317308

In [8]:
pd.crosstab(raw_data['Product'], columns = raw_data['Gender'], margins= True, normalize=True)

Gender     Female      Male       All
Product                              
KP281    0.222222  0.222222  0.444444
KP481    0.161111  0.172222  0.333333
KP781    0.038889  0.183333  0.222222
All      0.422222  0.577778  1.000000